web-module/src/main/webapp/templates/celAjax/struct/Tiny4Config.vm

```diff
@@ -129,52 +129,78 @@ $!services.rteconfig.setRteConfigHint("tinymce4")
 #set($filebaseURL = "$!{filebaseURL}&src_doc=${filePickerSpace}.${filePickerDoc}&columns=10")
 #set($filebaseURL = "$!{filebaseURL}&root=${filePickerSpace}&hasUpload=$hasUpload")
 ##
+## lookup the rte_styles field in the XWiki and WebPreference
+#set($rteStyles = $services.rteconfig.getRTEConfigField('styles'))
+## get multilingual style names
+#set($mlRteStyleArray = [])
+#set($mlRteStyleName = '')
+#foreach($rteStyleElement in $rteStyles.split('[=;]'))
+  #if($velocityCount%2 == 1)
+    #set($mlRteStyleName = $adminMsg.get("$rteStyleElement"))
+  #elseif("$mlRteStyleName" != '')
+    #set($mlRteStyleEntry = {})
+    #set($!stfu = $mlRteStyleEntry.put("title","${mlRteStyleName}"))
+    #set($!stfu = $mlRteStyleEntry.put("inline","span"))
+    #set($!stfu = $mlRteStyleEntry.put("classes","${rteStyleElement}"))
+    #set($!stfu = $mlRteStyleArray.add($mlRteStyleEntry))
```

Comment:
you could aready use `JsonBuilder` here and skip the creation of a `List<Map>>` entirely.
`JsonBuilder#addProperty` accepts `JsonBuilder` objects, like:
`$jsonBuilder.addProperty("style_formats", $mlRteStyleBuilder)`